app/controllers/import_controller.rb

```diff
@@ -38,108 +46,139 @@ def import
+              @errors << ValidationError.new(error, row_ind).message
             end
-
           end
+        end
 
-          # Checking for duplicates under current user's viewable patients is acceptable because custom jurisdictions must fall under hierarchy
-          patient[:duplicate_data] = patients_for_duplicate_detection.duplicate_data_detection(patient)
-        rescue ValidationError => e
-          @errors << e&.message || "Unknown error on row #{row_ind}"
-        rescue StandardError => e
-          @errors << e&.message || 'Unexpected error'
+        # Validate using Patient model validators without saving
+        validation_patient = Patient.new(patient.slice(*Patient.attribute_names.map(&:to_sym)))
+        unless validation_patient.valid?(:import)
+          format_model_validation_errors(validation_patient).each do |error|
+            @errors << ValidationError.new(error, row_ind).message
+          end
         end
+
+        # Checking for duplicates under current user's viewable patients is acceptable because custom jurisdictions must fall under hierarchy
+        patient[:duplicate_data] = patients_for_duplicate_detection.duplicate_data_detection(patient)
+
         @patients << patient
       end
     rescue ValidationError => e
-      @errors << e&.message || "Unknown error on row #{row_ind}"
-    rescue Zip::Error
+      @errors << e&.message || "Unknown error on row #{row_ind + 1}"
+    rescue Zip::Error, FileFormatError
       # Roo throws this if the file is not an excel file
-      @errors << 'File Error: Please make sure that your import file is a .xlsx file.'
+      @errors << "File Error: Please make sure that your import file is a .#{format == :epix ? 'csv' : 'xlsx'} file."
     rescue ArgumentError, NoMethodError
       # Roo throws this error when the columns are not what we expect
-      @errors << 'Format Error: Please make sure that .xlsx import file is formatted in accordance with the formatting guidance.'
+      @errors << "Format Error: Please make sure that .#{format == :epix ? 'csv' : 'xlsx'} import file is formatted in accordance with the formatting guidance."
     rescue StandardError => e
       # This is a catch all for any other unexpected error
-      @errors << "Unexpected Error: '#{e&.message}' Please make sure that .xlsx import file is formatted in accordance with the formatting guidance."
+      @errors << "Unexpected Error: '#{e&.message}' Please make sure that .#{format == :epix ? 'csv' : 'xlsx'} import file is formatted in accordance with the"\
+                 ' formatting guidance.'
     end
 
     render json: { patients: @patients, errors: @errors }
   end
 
+  private
+
+  def import_sara_alert_format_field(patient, field, row, row_ind, col_num, workflow, valid_jurisdiction_ids)
+    if field == :jurisdiction_path
+      patient[:jurisdiction_id], patient[:jurisdiction_path] = validate_jurisdiction(row[95], row_ind, valid_jurisdiction_ids)
```

Comment:
Instead of `row[95]` should this be:
```ruby
row[SARA_ALERT_FORMAT_FIELDS.index(:jurisdiction_path)]```

Reply:
https://github.com/SaraAlert/SaraAlert/pull/884/commits/3457e9b4d07e754512ed75ae1d943abc67f10577#diff-ee31e076a80cc9f48da0394c0cb066d59bf2f8c68f68eb62cb5119d01712d90eR145